core/ir/riscv64/instr.c

```diff
@@ -42,9 +42,9 @@
 bool
 instr_set_isa_mode(instr_t *instr, dr_isa_mode_t mode)
 {
-    if (mode != DR_ISA_RV64IMAFDC)
+    if (mode != DR_ISA_RV64IMAFDC && mode != DR_ISA_SYNTHETIC)
```

Comment:
Wondering whether "synthetic" needs a qualifier in case we add a 2nd type of synthetic ISA...but not sure I have a good proposal: DR_ISA_REGONLY?  The "register-only" ISA?

Reply:
Category is also an important piece of info that is encoded.
Although if I had to pick one, register dependencies would be my choice as well.
DR_ISA_REGONLY would work for me.
Although, perhaps we want to postpone this decision when we have a need for an additional synthetic ISA?
I think this would just be a simple renaming of the enum and the `core/ir/synthetic` dir, there shouldn't be any "more fundamental" change involved.

Reply:
The problem is this is a public interface so we can't easily rename things: that breaks compatibility.  Maybe ask for some opinions offline and pick a permanent name.

Reply:
DR_ISA_REGDEPS is the winner.